### PR TITLE
Remove float in favor of flex

### PR DESCRIPTION
**Pull request by whmii**
An option to remove float entirely. I originally was not in favor of such a move. I have found that given the fact that the grid objects are already so tightly constrained, this is actually a rather seamless transition. 

Through the use of `display: flex;` and `flex-wrap: wrap;`, the grid columns maintain a very consistent 'feel' that behaves very similarly to floats. Screenshot attached.

screen shot 2016-03-01 at 3 37 58 pm

### CODE REVIEW

File: contrib/patterns/_grid.scss [hunk 1]
```diff
@@ -3,7 +3,7 @@
 }
 
 .grid__column {
-  @include grid-column;
+  @include grid-column();
```
Comment (wardpenney) [inline contrib/patterns/_grid.scss]: I thought our style guide discourages parentheses when no params are passed.

**Comment by wardpenney** [comment]
Been reading the non-normative spec on [flex items](https://www.w3.org/TR/css-flexbox-1/#flex-items), and this might actually work.

When `display: flex` is on, it makes each child a flex item. I am looking into the `width` thing now. It might work!

**Comment by tysongach** [comment]
@whmii Mind providing a quick explanation on closing this, for context later?

**Comment by whmii** [comment]
I think this is an interesting proof of concept but I feel that a system where grid objects operate independent of their container, not because of their container, would be more effective.

**Pull request closed without merging**